syft/execution/protocol.py

```diff
@@ -203,6 +203,7 @@ def assign(self, role_id, worker):
         """ Assign a worker to the specified role.
         """
         self.role_assignments.assign(role_id, worker)
```

Comment:
Seems like this assumes one worker per role?

Reply:
(It’s actually the line below.)

Reply:
Hum, indeed... Do you think we'd need to create a class for an instantiated `Role`? I hope not but I don't see any solution right now

Reply:
Conceptually, it should be that the worker has a role, rather than the role having a worker (which is a simple way to make tracing work, but shouldn’t be used for execution.) If we introduce an `Execution` class, those are basically instantiated `Roles`, so maybe that would solve this issue?

Reply:
Yep, I started thinking about `Executions` and was about to say that the problem doesn't exist so don't mind what I said, I talked before thinking 😅